lib/datasource.js

```diff
@@ -2594,6 +2594,54 @@ DataSource.prototype.ping = function(cb) {
   return cb.promise;
 };
 
+/**
+ * Execute an arbitrary command. The commands are connector specific,
+ * please refer to the documentation of your connector for more details.
+ *
+ * @param command String|Object The command to execute, e.g. an SQL query.
+ * @param [args] Array Parameters values to set in the command.
+ * @param [options] Object Additional options, e.g. the transaction to use.
+ * @returns Promise A promise of the result
+ */
+DataSource.prototype.execute = function(command, args = [], options = {}) {
```

Comment:
@bajtos I found the command type here a bit confusing: the function types defined at the bottom of this PR only takes `string` type as command. Any particular reason to allow an object?

Reply:
> Any particular reason to allow an object?

Some community connectors are using object value for the comman and I want to preserve support for those connectors.

See e.g. https://www.npmjs.com/package/loopback-connector-neo4j-graph

```js
var cypher = {
    query: "MATCH (u:User {email: {email}}) RETURN u",
    params: {
        email: 'alice@example.com',
    }
};
// ...
ModelClass.dataSource.connector.execute(
    cypher,
    [],
    options,
    callback
);
```

I think the connector authors misunderstood how command & args are meant to be used together. I think the following would be a better usage:

```ts
ModelClass.dataSource.connector.execute(
  `MATCH (u:User {email: {email}}) RETURN u`,
   {email: 'alice@example.com'},
   options,
   callback);
```

Based on this analysis, I am going to change the type of `args` from array to object. (Note that arrays are objects too.)